### PR TITLE
TF style guide

### LINKED ISSUE #424: Switching to TensorFlow style guide
We should consider switching to the [TensorFlow style guide](https://www.tensorflow.org/community/contribute/code_style).  It's essentially the same as Google's style guide for Python, with a few exceptions.  The rest of the TensorFlow ecosystem already follows this guide for their Python code.  Here's an incomplete list of some things that would change:
- PEP8 linting with pylint instead of flake8 (they provide a specific [pylintrc file](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/pylintrc) to use for checking code quality that we can also use)
- Files with unit tests live inside the modules they test (#410)
- Switching the interpreted tab length from 4 spaces to 2 spaces
- Line length reduced to 79 from 99 (made easier by the reduction in tab length)

Someone should be assigned this and own it if we decide to proceed.  I'd be willing to volunteer if nobody finds fixing linter errors particularly engaging :D

### PR DESCRIPTION
**Pull request by jvmancuso**
Closes #424 

Still have to continue linting files in `tf_encrypted/tensor/`, but got stuck on a bug where all tests are now failing with the following error:

```
_________________________________________________________________________________________________ TestTruncate.test_noninteractive_truncate __________________________________________________________________________________________________

self = <tests.test_truncate.TestTruncate testMethod=test_noninteractive_truncate>

    def test_noninteractive_truncate(self):
    
      prot = tfe.protocol.Pond(
          tensor_factory=int100factory,
          fixedpoint_config=fixed100_ni,
      )
    
      with tfe.Session() as sess:
    
        expected = np.array([12345.6789])
    
        w = prot.define_private_variable(
            expected * prot.fixedpoint_config.scaling_factor)  # double precision
        v = prot.truncate(w)  # single precision
    
>       sess.run(tf.global_variables_initializer())

tests/test_truncate.py:52: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tf_encrypted/session.py:111: in run
    feed_dict=feed_dict
../anaconda/envs/tf-encrypted/lib/python3.5/site-packages/tensorflow/python/client/session.py:929: in run
    run_metadata_ptr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tf_encrypted.session.Session object at 0x1270c8630>, handle = None, fetches = <tf.Operation 'init' type=NoOp>, feed_dict = None, options = None, run_metadata = None

    def _run(self, handle, fetches, feed_dict, options, run_metadata):
      """Perform either run or partial_run, depending the presence of `handle`."""
    
      def _feed_fn(feed, feed_val):
        for tensor_type, _, feed_fn, _ in _REGISTERED_EXPANSIONS:
          if isinstance(feed, tensor_type):
            return feed_fn(feed, feed_val)
        raise TypeError('Feed argument %r has invalid type %r' % (feed,
                                                                  type(feed)))
    
      # Check session.
      if self._closed:
>       raise RuntimeError('Attempted to use a closed Session.')
E       RuntimeError: Attempted to use a closed Session.

../anaconda/envs/tf-encrypted/lib/python3.5/site-packages/tensorflow/python/client/session.py:1075: RuntimeError
```

If anyone has any idea of what's broken here, I'd love to hear it :D

### CODE REVIEW

File: tf_encrypted/layers/activation.py [hunk 1]
```diff
@@ -1,91 +1,89 @@
...
 
-    """
-    Relu Layer
+  def initialize(self, *args, **kwargs) -> None:
+    pass
 
-    :See: tf.nn.relu
+  def forward(self, x):
     """
+    :param ~tensorflow_encrypted.protocol.pond.PondTensor x: The input tensor
```
Comment (mortendahl) [inline tf_encrypted/layers/activation.py]: `tf_encrypted` instead of `tensorflow_encrypted`

Comment (mortendahl) [reply tf_encrypted/layers/activation.py]: there may be others :)

File: bin/run [hunk 2]
```diff
@@ -17,9 +17,9 @@ enable_stats_monitor = os.environ.get('TFE_MONITOR_STATS') is not None
 enable_traces = os.environ.get('TFE_TRACE') is not None
 enable_debug = os.environ.get('TFE_DEBUG') is not None
 
-tfe.setMonitorStatsFlag(enable_stats_monitor)
-tfe.setTFETraceFlag(enable_traces)
-tfe.setTFEDebugFlag(enable_debug)
+tfe.set_tfe_stats_flag(enable_stats_monitor)
```
Comment (mortendahl) [inline bin/run]: if we're renaming then let's use a better matching name, say `set_tfe_events_flag` (per https://www.tensorflow.org/guide/summaries_and_tensorboard)

File: bin/serve [hunk 3]
```diff
@@ -40,49 +30,59 @@ enable_debug = os.environ.get('TFE_DEBUG') is not None
...
+    print('using fixed100 interactive truncation')
+    return fixed.fixed100
+  print('using fixed100 non interactive truncation')
+  return fixed.fixed100_ni
 
 
 if node_name is None:
-    print("The node_name of this node must be provided via the 'TFE_NODE_NAME' environment variable")
-    sys.exit(1)
+  err = ("The node_name of this node must be provided via"
```
[the remark this answers is not in the document]
Comment (jvmancuso) [reply bin/serve]: not really sure, was mainly doing this to keep to stay in line length and hopefully keeping things readable, maybe there's a better way

File: operations/secure_random/test_secure_random.py [hunk 4]
```diff
@@ -10,116 +11,123 @@
...
-    random_uniform = secure_random_module.secure_random_uniform
-    seed = secure_random_module.secure_seed
+  secure_random_module = tf.load_op_library(shared_object)
+  seeded_random_uniform = secure_random_module.secure_seeded_random_uniform
+  random_uniform = secure_random_module.secure_random_uniform
+  seed = secure_random_module.secure_seed
 except NotFoundError:
-    secure_random_module = None
+  secure_random_module = None
+  notfound_msg = "secure_random_module not found"
```
[the remark this answers is not in the document]
Comment (jvmancuso) [reply operations/secure_random/test_secure_random.py]: will give it a go

**Comment by jvmancuso** [comment]
closing this in favor of #474

**Pull request closed without merging**